File: dart/cell.dart
class Cell {
  final int x;
  final int y;
  bool alive;
  bool? nextState;
  final List<Cell> neighbours = [];

  Cell(this.x, this.y, [this.alive = false]);

  String toChar() {
    return alive ? 'o' : ' ';
  }

  int aliveNeighbours() {
    // The following is slower
    // return neighbours.where(
    //   (neighbour) => neighbour.alive
    // ).length;

    // The following is slower
    // var aliveNeighbours = 0;
    // neighbours.forEach((neighbour) {
    //   if (neighbour.alive) {
    //     aliveNeighbours += 1;
    //   }
    // });
    // return aliveNeighbours;

    // The following is the fastest
    var aliveNeighbours = 0;
    final count = neighbours.length;
    for (var i = 0; i < count; i++) {
      final neighbour = neighbours[i];
      if (neighbour.alive) {
        aliveNeighbours += 1;
      }
    }
    return aliveNeighbours;
  }
}

File: c#/cell.cs
using System.Collections.Generic;
// using System.Linq;

public class Cell(uint x, uint y, bool alive = false) {
  public readonly uint X = x;
  public readonly uint Y = y;
  public bool Alive = alive;
  public bool? NextState = null;
  public readonly List<Cell> Neighbours = [];

  public char ToChar() {
    return this.Alive ? 'o' : ' ';
  }

  public uint AliveNeighbours() {
    // The following is slower
    // return (uint)this.Neighbours.Where(
    //   (neighbour) => neighbour.Alive
    // ).ToList().Count;

    // The following is slower
    // var aliveNeighbours = 0u;
    // foreach (var neighbour in this.Neighbours) {
    //   if (neighbour.Alive) {
    //     aliveNeighbours++;
    //   }
    // }
    // return aliveNeighbours;

    // The following is the fastest
    var aliveNeighbours = 0u;
    var count = this.Neighbours.Count;
    for (var i = 0; i < count; i++) {
      var neighbour = this.Neighbours[i];
      if (neighbour.Alive) {
        aliveNeighbours++;
      }
    }
    return aliveNeighbours;
  }
}
